### PR TITLE
Update @vaticle_dependencies

### PR DESCRIPTION
**Pull request by lolski**
## What is the goal of this PR?

We updated the version of `@vaticle_dependencies` to the latest one. Invocations of various build toolings have also been updated accordingly.

## What are the changes implemented in this PR?

1. Update the version of `vaticle_dependencies`.
2. Update other repo versions to the ones that have updated `@vaticle_dependencies` as well.
3. Update import statements and invocations of `createnotes:bin` and `release_validate_deps`

### CODE REVIEW

File: .grabl/automation.yml [hunk 1]
```diff
@@ -178,8 +178,8 @@ release:
         pyenv install -s 3.6.10
         pyenv global 3.6.10 system
         pip install certifi
-        export CREATE_NOTES_TOKEN=$REPO_GITHUB_TOKEN
-        bazel run @vaticle_dependencies//tool/release/createnotes:bin -- $GRABL_OWNER $GRABL_REPO $GRABL_COMMIT $(cat VERSION) ./RELEASE_TEMPLATE.md
+        export NOTES_CREATE_TOKEN=$REPO_GITHUB_TOKEN
```
Comment (lolski) [inline .grabl/automation.yml:181]: Update create-notes target.

File: BUILD [hunk 2]
```diff
@@ -233,7 +233,9 @@ checkstyle_test(
 filegroup(
     name = "ci",
     data = [
+        "@vaticle_dependencies//factory/analysis:dependency-analysis",
         "@vaticle_dependencies//library/maven:update",
+        "@vaticle_dependencies//tool/release/notes:create",
```
Comment (lolski) [inline BUILD:238]: Update create-notes target.

File: BUILD [hunk 3]
```diff
@@ -233,7 +233,9 @@ checkstyle_test(
 filegroup(
     name = "ci",
     data = [
+        "@vaticle_dependencies//factory/analysis:dependency-analysis",
```
Comment (lolski) [inline BUILD:236]: Add a `dependency-analysis` since it is called on the CI.

File: BUILD [hunk 4]
```diff
@@ -25,7 +25,7 @@ load("@vaticle_dependencies//builder/java:rules.bzl", "native_java_libraries")
 load("@vaticle_dependencies//distribution:deployment.bzl", "deployment")
 load("@vaticle_dependencies//distribution/artifact:rules.bzl", "artifact_repackage")
 load("@vaticle_dependencies//tool/checkstyle:rules.bzl", "checkstyle_test")
-load("@vaticle_dependencies//tool/release:rules.bzl", "release_validate_deps")
+load("@vaticle_dependencies//tool/release/deps:rules.bzl", "release_validate_deps")
```
Comment (lolski) [inline BUILD:28]: Update the import statement of `release_validate_deps` to reflect the new path.

File: dependencies/vaticle/repositories.bzl [hunk 5]
```diff
@@ -21,40 +21,40 @@ def vaticle_dependencies():
     git_repository(
         name = "vaticle_dependencies",
         remote = "https://github.com/vaticle/dependencies",
-        commit = "4d6ca1b6d7fe93e4c11b18aeba5d5889b928aadd", # sync-marker: do not remove this comment, this is used for sync-dependencies by @vaticle_dependencies
+        commit = "aa6eb2eab10b032c0ead898b0a5eead1bea057c0", # sync-marker: do not remove this comment, this is used for sync-dependencies by @vaticle_dependencies
```
Comment (lolski) [inline dependencies/vaticle/repositories.bzl:24]: Update the version of `@vaticle_dependencies`. Consequently, update other repo versions to the ones that have updated `@vaticle_dependencies` as well.

**Review by flyingsilverfin — APPROVED** [review]
Looks good!! Thanks :)

**Pull request merged**